sdk/servicebus/service-bus/test/retries.spec.ts

```diff
@@ -342,3 +343,96 @@ describe("Retries - MessageSender", () => {
     });
   });
 });
+
+describe("Retries - Receive methods", () => {
+  let receiverClient: Receiver<ReceivedMessageWithLock>;
+  let serviceBusClient: ServiceBusClientForTests;
+  let defaultMaxRetries = 2;
+  let numberOfTimesTried: number;
+
+  before(() => {
+    serviceBusClient = createServiceBusClientForTests({
+      retryOptions: {
+        // Defaults
+        timeoutInMs: 10000,
+        maxRetries: defaultMaxRetries,
+        retryDelayInMs: 0
+      }
+    });
+  });
+
+  after(() => {
+    return serviceBusClient.test.after();
+  });
+
+  async function beforeEachTest(entityType: TestClientType): Promise<void> {
+    const entityNames = await serviceBusClient.test.createTestEntities(entityType);
+    receiverClient = serviceBusClient.test.getPeekLockReceiver(entityNames);
+  }
+
+  async function afterEachTest(): Promise<void> {
+    await receiverClient.close();
+  }
+
+  function mockBatchingReceiveToThrowError() {
+    const fakeFunction = async function() {
+      numberOfTimesTried++;
+      throw new MessagingError("Hello there, I'm an error");
+    };
+    // Mocking batchingReceiver.receive to throw the error and fail
+    const batchingReceiver = BatchingReceiver.create((receiverClient as any)._context);
+    batchingReceiver.isOpen = () => true;
+    batchingReceiver.receive = fakeFunction;
+    // Mocking session creation to throw the error and fail
+    (receiverClient as any)._createMessageSessionIfDoesntExist = fakeFunction;
```

Comment:
DUDE...NICE!

Reply:
Hey, if these are purely unit tests toss them under the `internal` folder.

Reply:
Actually, I tried when you've suggested in the other PR, the test-utils is being used in this file, which looks for the environment variables like connection string, and they are not available for unit-tests. So, just left these in here.

Reply:
I might update the test-utils later on to handle these cases and make them pure unit-tests.

Reply:
Okay, no biggie. :)